.github/workflows/automerge.yml

```diff
@@ -5,11 +5,35 @@
 # This reduces the friction associated with updating with our workflows.
 
 on: [ pull_request ]
+name: Automerge
 
 jobs:
-  automerge:
+  automerge-check:
     if: github.event.pull_request.user.login == 'web3-bot'
     runs-on: ubuntu-latest
+    outputs:
+      status: ${{ steps.should-automerge.outputs.status }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Check if we should automerge
+      id: should-automerge
+      run: |
+        for commit in $(git rev-list --first-parent origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.sha }}); do
+          committer=$(git show --format=$'%ce' -s $commit)
+          echo "Committer: $committer"
+          if [[ "$committer" != "web3-bot@users.noreply.github.com" ]]; then
+            echo "Commit $commit wasn't committed by web3-bot, but by $committer."
```

Comment:
I was wondering; do we follow any specific bash style guide like [this](https://google.github.io/styleguide/shellguide.html)?

Reply:
Not that I'm aware. This is all linside YAML, so trying to follow a strict format is going to be tricky...

Reply:
I agree; bash in yaml is, well, bash in yaml 😊

Reply:
better take that up over in https://github.com/protocol/.github I think